javascript/MaterialXView/source/viewer.js

```diff
@@ -769,31 +926,108 @@ export class Material
+                img.src = 'public/shader_ball.svg';
+                //matTitle.classList.remove('peditor_material_unassigned');
+            }
+        }
+    }
+
+    static updateSoloMaterial(viewer, elemPath, materials, event)
+    {
+        // Prevent the event from being passed to parent folder
+        event.stopPropagation();
+
+        for (let i=0; i<materials.length; ++i)
+        {
+            let matassign = materials[i];
+            // Need to use path vs name to get a unique key.
+            let materialName = matassign.getMaterial().getNamePath();
+            var matUI = matassign.getMaterialUI();
+            let matTitle = matUI.domElement.getElementsByClassName('title')[0];
+            let img = matTitle.getElementsByTagName('img')[0];    
+            if (materialName == elemPath)
+            {                
+                if (this._soloMaterial == elemPath)
+                {
+                    img.src = 'public/shader_ball.svg';
+                    matTitle.classList.remove('peditor_material_assigned');
+                    this._soloMaterial = "";
+                }
+                else
+                {
+                    img.src = 'public/shader_ball2.svg';
+                    matTitle.classList.add('peditor_material_assigned');
+                    this._soloMaterial = elemPath;
+                }
+            }
+            else
+            {
+                img.src = 'public/shader_ball.svg';
+                matTitle.classList.remove('peditor_material_assigned');                   
+            }
+        }
+        viewer.getMaterial().updateMaterialAssignments(viewer, this._soloMaterial);
+        viewer.getScene().setUpdateTransforms();
+    }
+
     //
     // Update property editor for a given MaterialX element, it's shader, and
     // Three material
     //
-    updateEditor(elem, shader, material, gui, viewer, closeUI)
+    updateEditor(matassign, shader, material, gui, closeUI, viewer)
     {
+        var elem = matassign.getMaterial();
+        var materials = this._materials;
+
         const DEFAULT_MIN = 0;
         const DEFAULT_MAX = 100;
 
         var startTime = performance.now();
 
         const elemPath = elem.getNamePath();
-        var matUI = gui.addFolder(elemPath + ' Properties');
+
+        // Create and cache associated UI
+        var matUI = gui.addFolder(elemPath);        
+        matassign.setMaterialUI(matUI);
+
+        let matTitle = matUI.domElement.getElementsByClassName('title')[0];
+        // Add a icon to the title to allow for assigning the material to geometry
+        // Clicking on the icon will "solo" the material to the geometry.
+        // Clicking on the title will open/close the material folder.
+        matTitle.innerHTML = "<img id='" + elemPath + "' src='public/shader_ball.svg' width='16' height='16' style='vertical-align:middle; margin-right: 5px;'>" + elem.getNamePath();
```

Comment:
This mechanism of inserting additional DOM elements + event handler allows for insertion of additional interaction. We can do this anywhere we'd like to add in additional action "buttons".